src/test/java/org/broadinstitute/hellbender/tools/exome/allelefraction/AlleleFractionModellerUnitTest.java

```diff
@@ -0,0 +1,319 @@
+        // more hets -- most of the error is the sampling error of a finite simulated data set, not numerical error of MCMC
+        final double minorFractionTolerance = 0.03;
+        final double meanBiasTolerance = 0.03;
+        final double biasVarianceTolerance = 0.04;
+        final double outlierProbabilityTolerance = 0.03;
+        AlleleFractionSimulatedData simulatedData = new AlleleFractionSimulatedData(averageHetsPerSegment, numSegments,
+                averageDepth, meanBias, biasVariance, outlierProbability);
+
+        final AlleleFractionModeller model = new AlleleFractionModeller(simulatedData.getSegmentedModel());
+        model.fitMCMC(numSamples, numBurnIn);
+
+
+        final List<Double> meanBiasSamples = model.getmeanBiasSamples();
+        Assert.assertEquals(meanBiasSamples.size(), numSamples - numBurnIn);
+
+        final List<Double> biasVarianceSamples = model.getBiasVarianceSamples();
+        Assert.assertEquals(biasVarianceSamples.size(), numSamples - numBurnIn);
+
+        final List<Double> outlierProbabilitySamples = model.getOutlierProbabilitySamples();
+        Assert.assertEquals(outlierProbabilitySamples.size(), numSamples - numBurnIn);
+
+        final List<AlleleFractionState.MinorFractions> minorFractionsSamples = model.getMinorFractionsSamples();
+        Assert.assertEquals(minorFractionsSamples.size(), numSamples - numBurnIn);
+        for (final AlleleFractionState.MinorFractions sample : minorFractionsSamples) {
+            Assert.assertEquals(sample.size(), numSegments);
+        }
+
+        final List<List<Double>> minorFractionsSamplesBySegment = model.getMinorFractionSamplesBySegment();
+
+        final double mcmcMeanBias = meanBiasSamples.stream().mapToDouble(x -> x).average().getAsDouble();
+        final double mcmcBiasVariance = biasVarianceSamples.stream().mapToDouble(x -> x).average().getAsDouble();
+        final double mcmcOutlierProbabilityr = outlierProbabilitySamples.stream().mapToDouble(x -> x).average().getAsDouble();
+        final List<Double> mcmcMinorFractions = minorFractionsSamplesBySegment
+                .stream().map(list -> list.stream().mapToDouble(x -> x).average().getAsDouble())
+                .collect(Collectors.toList());
+
+        double totalSegmentError = 0.0;
+        for (int segment = 0; segment < numSegments; segment++) {
+            totalSegmentError += Math.abs(mcmcMinorFractions.get(segment) - simulatedData.getTrueState().minorFractionInSegment(segment));
+        }
+
+        Assert.assertEquals(mcmcMeanBias, meanBias, meanBiasTolerance);
+        Assert.assertEquals(mcmcBiasVariance, biasVariance, biasVarianceTolerance);
+        Assert.assertEquals(mcmcOutlierProbabilityr, outlierProbability, outlierProbabilityTolerance);
+        Assert.assertEquals(totalSegmentError / numSegments, 0.0, minorFractionTolerance);
+    }
+
+    @Test
+    public void testSamplers() {
+
+        final int RANDOM_SEED = 15;
```

Comment:
i think you might have addressed my comment to make these variable names uppercase when you had a separate class to test the Samplers, but as they are now they probably should be lowercase...sorry about that!